app/assets/javascripts/widgets/notifications-badge.js

```diff
@@ -79,31 +79,25 @@
       self.dropdownNotifications.empty();
       $.each(self.notifications, function(index, notifications) {
         $.each(notifications, function(index, notification) {
-          if($.inArray(notification, notifications) === -1)
+          if($.inArray(notification, notifications) === -1){
             self.dropdownNotifications.append(notification.note_html);
+          }
         });
       });
 
       app.helpers.timeago(self.dropdownNotifications);
 
-      self.dropdownNotifications.find('.unread').each(function() {
-        Diaspora.page.header.notifications.setUpUnread( $(this) );
-      });
-      self.dropdownNotifications.find('.read').each(function() {
-        Diaspora.page.header.notifications.setUpRead( $(this) );
-      });
-      $('.notifications').perfectScrollbar('destroy');
-      $('.notifications').perfectScrollbar();
+      self.dropdownNotifications.perfectScrollbar('destroy').perfectScrollbar();
       self.ajaxLoader.hide();
-      isLoading = false;
-      $('.notifications').removeClass("loading");
+      var isLoading = false;
+      self.dropdownNotifications.removeClass("loading");
       //Infinite Scrolling
-      $('.notifications').scroll(function() {
-        var bottom = $('.notifications').prop('scrollHeight') - $('.notifications').height();
-        var currentPosition = $('.notifications').scrollTop();
+      self.dropdownNotifications.scroll(function() {
+        var bottom = self.dropdownNotifications.prop('scrollHeight') - self.dropdownNotifications.height();
+        var currentPosition = self.dropdownNotifications.scrollTop();
         isLoading = ($('.loading').length === 1);
         if (currentPosition + 50 >= bottom && notificationsLoaded <= self.notifications.length && !isLoading) {
-            $('.notifications').addClass("loading");
+            self.dropdownNotifications.addClass("loading");
             ++currentPage;
             self.getMoreNotifications();
         }
```

Comment:
Maybe we can use the infinite scrolling plugin we already have instead? Or at least drop the existing one and use the new one everywhere? I would like to not add two plugins for essentially the same job.